### DISCUSSION TITLE
Basketball specific model, 3 models to 1 model

### DISCUSSION THREAD

**Discussion by mjerdem** [ultralytics/ultralytics#24030] [Q&A] [answered]
I have been training YOLO26 (Fast) model only to find basketball for indoor courts (much more specific than class 32)
Also training same, Yolo26 (fast) model for rim+net +ball to identify score
The remaining interest is to track the person. I can't do that because it is just me. I feel like it will only thin the model is trianing specifically for me. 

So I now have 2 custom models and using YOL026 (Class 0) for person. 

Is there a way to combine these models and end up with one file to run on the image, not 3 different models?  I know the concept of Transfer Learning but I am not sure if that is what I want and if it is how to do that?  I will be using these models, or hopefully the model, in an Android phone. 

Thank you

**Comment by UltralyticsAssistant** [comment]
👋 Hello @mjerdem, thank you for your interest in Ultralytics 🚀 This is an automated response to help get things moving, and an Ultralytics engineer will also assist you soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

Since this looks like a custom training ❓ question, please share as much information as possible so the team can guide you effectively, including:
- your current dataset structure and class list
- whether your basketball, rim+net, and person data come from the same images or different datasets
- a few dataset image examples
- your training configuration and logs
- the exact YOLO26 model variant you trained
- whether you want one model for detection only, or detection plus tracking in the same pipeline

Please also verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

If you run into any unexpected behavior or think there may be a 🐛 bug involved, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Reply by glenn-jocher** [reply]
Yes — but not by merging the three `.pt` files after training; the usual path is to build one combined detection dataset with classes like `person`, `basketball`, `rim`, and `net`, then fine-tune a single YOLO26 model on that unified data, while using built-in [tracking mode](https://docs.ultralytics.com/modes/track/) at inference to keep player IDs across frames, so you do **not** need to train a separate “person tracker” just on yourself; for Android, a single custom detector exported with the [deployment/export workflow](https://docs.ultralytics.com/guides/model-deployment-options/) is the cleanest setup.

**Reply by mjerdem** [reply]
Thank you for your answer but I am new to Ai training  so I am sorry if my followup questions are kind of "stupid".
The way I understand I need to train a model that can identify a person, basketball, rim, and net. I can't not just train basketball, rim and net (to identify make and miss) and use(Transfer learning) Class 0 (person) from YOLO26 pretrained data. Am I right? 

Second question is about tracking mode. Lets assume I have to train the model for what is person, basketball, rim, and net. Now I have a custom YOLO26 model can identiy all those when I am exporting that I will be using TF Lite so I can use in Android Studio. How can I include the tracking mode while exporting? 

Thank you

**Reply by glenn-jocher** [reply, marked as answer]
Yes, that’s right: if you want one `.pt`/`.tflite` file, fine-tune a single YOLO26 detector on one combined dataset with `person`, `basketball`, `rim`, and `net` (the images do **not** all need to contain all four classes); transfer learning here means starting from pretrained YOLO26 weights, not attaching COCO `person` to a separate 3-class model after training. Also, tracking is not exported inside TFLite — `track` is a runtime step built on top of detections, so on Android you export only the detector and handle ID tracking separately in the app; see the [modes overview](https://docs.ultralytics.com/modes/) and [`Model.track()` reference](https://docs.ultralytics.com/reference/engine/model/#ultralytics.engine.model.Model.track).
